src/CatalogApplication/Models/README.md

```diff
@@ -0,0 +1,54 @@
+# UML Model For Tags/Ratings
+![alt text](https://i.imgur.com/eMo0eor.png "UML Diagram")
+
+# Tag Attributes
+```json
+Tag(
+    id:          int      # Generated by us somehow
+    name:        str      # The actual tag name
+    color:       int      # Hex value
+    note_id:     int      # Note ID associated with this tag
+    hidden:      bool     # Should the tag be shown
+    is_system:   bool     # Was this tag automatically added by us?
+    category:    str      # Engineering, Chemistry, CS, etc?
+)
+```
+
+## unique_id
+This is an integer generated by +1 increment
+
+## name
+A string representing the literal name of the tag i.e data-structures
+
+## color
+Hexadecimal integer value representing "good" or "bad" tag
```

Comment:
Why did you choose to use a hex vs. boolean determining good or bad tag?

Reply:
I can speak on this one a little. At first we thought that each tag should have a color hex value associated with it and that the hash of the tag name would determine this. We might simplify our data down to just a boolean good/bad and let the UI folks determine what to do with that in terms of displaying it.